app/code/Magento/Inventory/Model/ReservationAppend.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\InputException;
+use Magento\Inventory\Model\ResourceModel\Reservation\SaveMultiple;
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+use Magento\InventoryApi\Api\ReservationAppendInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class ReservationAppend implements ReservationAppendInterface
+{
+    /**
+     * @param SaveMultiple $saveMultiple
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        SaveMultiple $saveMultiple,
+        LoggerInterface $logger
+    ) {
+        $this->saveMultiple = $saveMultiple;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $reservations)
+    {
+        /** @var ReservationInterface $reservation */
+        foreach ($reservations as $reservation) {
+            if (!is_null($reservation->getReservationId())) {
```

Comment:
The code checks whether reservation **is not** null; Reservations can only be appended so if you try to update an existing one (that is an entity with a not null ID) you get an exception.